crates/containerd-shim-wasm/src/sandbox/oci.rs

```diff
@@ -35,6 +35,36 @@ pub fn get_args(spec: &Spec) -> &[String] {
     }
 }
 
+// get_module returns the module name and exported function name to be called
+// from the arguments on the runtime spec process field. The first argument will
+// be the module name and the default function name is "_start".
+//
+// If there is a '#' in the argument it will split the string
+// returning the first part as the module name and the second part as the function
+// name.
+//
+// example: "module.wasm#function" will return (Some("module.wasm"), "function")
+//
+// If there are no arguments then it will return (None, "_start")
+pub fn get_module(spec: &Spec) -> (Option<String>, String) {
+    let args = get_args(spec);
+
+    if !args.is_empty() {
+        let start = args[0].clone();
+        let mut iterator = start.split('#');
+        let mut cmd = iterator.next().unwrap().to_string();
```

Comment:
I guess `#_start` is invalid module name?

Reply:
There is a little bit of a discussion in https://github.com/containerd/runwasi/pull/102 where this was originally introduced. 

My understanding is that this is really a runwasi specific seperator and could be anything (`::` was suggested at one point).  I scanned through https://github.com/WebAssembly/WASI but couldn't find anything that said `#` is valid or not

Reply:
I think that if this is a runwasi-specific thing, we could actually write down the semantics in how this should work and especially which entrypoint strings are valid and which not. Then we could validate the input string based on that. The benefit would be that other projects could use the same notation and semantics, and we could document the notation also for the users.

For example, should the following strings work (or not):

1. `foo#bar`
2. `#bar`
3. `foo#bar#bar`
4. `foo`
5. `#foo#bar`
6. `foo#`
7. `foo#bar; -- %`

Reply:
this is a good idea, are you ok with opening an issue to track this?  This particular PR isn't introducing anything new, just moving some code around so we don't have the same logic in several different places.

Reply:
Ok, added the issue.